src/Chainweb/TreeDB.hs

```diff
@@ -390,9 +298,10 @@ class (Typeable db, TreeDbEntry (DbEntry db)) => TreeDb db where
             -- Upper limits
         -> HS.HashSet (UpperBound (DbKey db))
             -- Lower limits
-        -> S.Stream (Of (DbKey db)) IO (Natural, Eos)
-    branchKeys db k l mir mar lower uppper = S.map key
-        $ branchEntries db k l mir mar lower uppper
+        -> (S.Stream (Of (DbKey db)) IO (Natural, Eos) -> IO a)
```

Comment:
Since this is so prevalent maybe a general comment + a type synonym?

Reply:
The type could use some further cleanup. I think, that with the changes in this PR, we may be able to remove the `Eos` parameter. Since that change would affect the REST API non-trivially, I propose to do it in a separate Task/PR.

Reply:
ok